internal/lm/nvml.go

```diff
@@ -193,3 +199,43 @@ func isMPSCapable(manager resource.Manager) (bool, error) {
 	}
 	return true, nil
 }
+
+func newGPUModeLabeler(manager resource.Manager) (Labeler, error) {
+
+	devices, err := manager.GetDevices()
+	if err != nil {
+		return nil, err
+	}
+	if len(devices) == 0 {
+		// no devices, return empty labels
+		return empty{}, nil
+	}
+
+	class := ""
+	// loop through all devices to check if all of them are on same gpu mode
+	for _, d := range devices {
+		val, err := d.GetClass()
+		if err != nil {
+			return nil, err
+		}
+		if class != "" && val != class {
+			break
+		} else {
+			class = val
+		}
+	}
```

Comment:
Let's move this to a different function that is a little more generic:
```
func getDevicePCIEClasses(devices []resource.Device) ([]string, error) {
	seenClasses := make(map[string]bool)
	for _, d := range devices {
		class, err := d.GetClass()
		if err != nil {
			return nil, err
		}
		seenClasses[class] = true
	}

	var classes []string
	for class := range seenClasses {
		classes = append(classes, class)
	}

	return classes, nil
}
```

We may also want to introduce a function that returns the mode from these classes (or for a slice of devices) and then use this to set the label value.

Reply:
Fixed this